File: Rohit Singhal/TwoSum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> map=new HashMap<>();
        int[] arr=new int[2];
        for(int i=0;i<nums.length;i++)
        {
            int comp=target-nums[i];
            if(map.containsKey(comp))
            {
                arr[0]=map.get(comp);
                arr[1]=i;
                return arr;
            }
            map.put(nums[i],i);
            
        }
     return arr;
        
    }
}

File: Indal Kumar/twoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        unordered_map<int,int> m;
        
        int n=nums.size();
        
        for(int i=0;i<n;i++){
            
            int num=nums[i];
            int c=target-num;
            auto it=m.find(c);
            
            if(it!=m.end()){
                return {it->second,i};
            }else{
                m[num]=i;
                
            }
        }
        return {};
        
    }
};
